hls4ml/backends/vivado/passes/conv_stream.py

```diff
@@ -13,21 +15,25 @@ def transform(self, model, node):
         elif '2D' in node_class:
             self._generate_2d_instructions(node)
         else:
-            raise Exception('Cannot generate instructions for node {} ({})'.format(node.name, node_class))
-    
+            raise Exception(f'Cannot generate instructions for node {node.name} ({node_class})')
+
     def _generate_1d_instructions(self, node):
         if node.model.config.get_config_value('IOType') == 'io_stream':
             min_w, instructions = node.model.config.backend.compute_conv1d_instructions(
                 node.get_input_variable().shape[0],
                 node.get_input_variable().shape[1],
                 node.get_attr('filt_width'),
-                node.get_attr('stride_width'))
+                node.get_attr('stride_width'),
+            )
             instructions_str = ','.join(str(i) for i in instructions)
             node.set_attr('min_width', min_w)
             node.set_attr('instructions', instructions_str)
+            node.set_attr('unscaled', node.get_input_variable().shape[0] == min_w)
```

Comment:
Do you really need an attribute here? Seems like a the check can also be done in the template itself. Everything can be an attribute, true, but less attributes makes it easier to keep track of them

Reply:
I don't really have a strong opinion. I thought it was easier to define an attribute, to make the logic explicit, though I didn't think too much. Let me look at it once more.

Reply:
I agree with your proposal. I went ahead and pushed it.